File: backend/src/main/java/com/ironlogic/modules/workout/domain/model/WorkoutExercise.java
package com.ironlogic.modules.workout.domain.model;

import java.time.LocalDateTime;

/**
 * WorkoutSession 中单个训练动作的领域模型。
 *
 * <p>WorkoutExercise 可以来自模板复制，也可以在训练过程中手动新增。
 *
 * @param id WorkoutExercise id
 * @param workoutSessionId 所属 WorkoutSession id
 * @param exerciseId 引用的 Exercise id
 * @param sourceTemplateExerciseId 可选来源模板动作 id
 * @param actualOrderNo 在训练中的实际顺序
 * @param replacementOfExerciseId 预留给未来动作替换逻辑的字段
 * @param notes 可选动作备注
 * @param createdAt 创建时间
 * @param updatedAt 更新时间
 */
public record WorkoutExercise(
        Long id,
        Long workoutSessionId,
        Long exerciseId,
        Long sourceTemplateExerciseId,
        Integer actualOrderNo,
        Long replacementOfExerciseId,
        String notes,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {
}

File: frontend/lib/features/workout/domain/model/workout_exercise.dart
import 'workout_set.dart';

/// WorkoutExercise 领域模型。
///
/// 该类属于 `workout/domain`，表示一次训练中的一个动作记录。
class WorkoutExercise {
  /// 创建 WorkoutExercise 模型。
  const WorkoutExercise({
    required this.id,
    required this.workoutSessionId,
    required this.exerciseId,
    required this.actualOrderNo,
    required this.sets,
    this.sourceTemplateExerciseId,
    this.replacementOfExerciseId,
    this.notes,
    this.createdAt,
    this.updatedAt,
  });

  /// 动作记录 id。
  final int id;

  /// 所属训练 id。
  final int workoutSessionId;

  /// 引用的 Exercise id。
  final int exerciseId;

  /// 来源模板动作 id。
  final int? sourceTemplateExerciseId;

  /// 实际顺序号。
  final int actualOrderNo;

  /// 替换来源动作 id。
  final int? replacementOfExerciseId;

  /// 备注。
  final String? notes;

  /// 创建时间。
  final DateTime? createdAt;

  /// 更新时间。
  final DateTime? updatedAt;

  /// 组列表。
  final List<WorkoutSet> sets;
}
